src/main/java/com/bmuschko/gradle/docker/tasks/image/DockerCommitImage.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.bmuschko.gradle.docker.tasks.image;
+
+import com.bmuschko.gradle.docker.internal.RegularFileToStringTransformer;
+import com.bmuschko.gradle.docker.tasks.container.DockerExistingContainer;
+import com.github.dockerjava.api.command.CommitCmd;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.OutputFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+public class DockerCommitImage extends DockerExistingContainer {
+
+    /**
+     * The repository and image name e.g. {@code vieux/apache}.
+     *
+     * @since 9.0.0
+     */
+    @Input
+    public final Property<String> getRepository() {
+        return repository;
+    }
+
+    private final Property<String> repository = getProject().getObjects().property(String.class);
```

Comment:
For future reference, can you organize all fields together and then the getter methods together?

Reply:
Yes, will do going forward.